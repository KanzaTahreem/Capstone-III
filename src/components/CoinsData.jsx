import React from 'react';
import PropTypes from 'prop-types';

const CoinsData = (props) => {
  const { image, name } = props;
  return (
    <div className="inner-card">
      <img
        src={image}
        alt={`${image}-logo`}
        className="card-img"
      />
      <p className="card-name">{name}</p>
    </div>
  );
};

CoinsData.propTypes = {
  image: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
};

export default CoinsData;
