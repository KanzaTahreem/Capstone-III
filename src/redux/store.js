import { configureStore } from '@reduxjs/toolkit';
import coinsReducer from './AllCoins/coins';
import chartReducer from './CoinChart/chart';

const store = configureStore({
  reducer: {
    crypto: coinsReducer,
    chart: chartReducer,
  },
});

export default store;
