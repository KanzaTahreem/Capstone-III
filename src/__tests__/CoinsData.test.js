import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../redux/store';
import CoinsData from '../components/CoinsData';

it('renders correctly', () => {
  const rocket = renderer
    .create(<Provider store={store}><CoinsData /></Provider>)
    .toJSON();
  expect(rocket).toMatchSnapshot();
});
