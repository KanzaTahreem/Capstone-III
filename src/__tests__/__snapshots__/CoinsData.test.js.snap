// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="inner-card"
>
  <img
    alt="undefined-logo"
    className="card-img"
  />
  <p
    className="card-name"
  />
</div>
`;
